observability/telemetry/src/main/scala/com/daml/telemetry/TelemetryContext.scala

```diff
@@ -168,6 +170,11 @@ protected class DefaultTelemetryContext(protected val tracer: Tracer, protected
   }
 
   override def openTelemetryContext: Context = Context.current.`with`(span)
+
+  lazy val traceId: Option[String] = Option(Span.fromContextOrNull(this.openTelemetryContext))
+    .filter(_.getSpanContext.isValid)
+    .map(_.getSpanContext.getTraceId)
```

Comment:
Is this the traceId that is used in Canton as well? It could help if you could add a brief description of what you aim to solve here to the PR or to the ticket otherwise it's harder to review.

Reply:
Updated the description.

Reply:
Thanks! I see we already have a `Span` in scope (argument to the `DefaultTelemetryContext`). Why don't we use this one? Also, is there a specific reason on why we want to have `traceId` lazy?

Reply:
You are right I will use the existent `Span`.